JetStreamCompose/jetstream/src/main/java/com/google/jetstream/data/repositories/MovieRepositoryImpl.kt

```diff
@@ -33,132 +33,167 @@ import com.google.jetstream.data.util.StringConstants.Movie.Reviewer.DefaultCoun
+                }
             )
         }
-        emit(MovieList(value = list))
     }
 
-    override fun getTop10Movies(): Flow<MovieList> = flow {
-        val list = top250MoviesWithWideThumbnail.subList(fromIndex = 20, toIndex = 30)
-        emit(MovieList(value = list))
+    override fun getFeaturedMovies(): Flow<MovieList> = top250MoviesWithWideThumbnail.map {
+        val list = it.filterIndexed { index, _ -> listOf(1, 3, 5, 7, 9).contains(index) }
+        MovieList(list)
     }
 
-    override fun getNowPlayingMovies(): Flow<MovieList> = flow {
-        val list = inTheatersMovies
+    override fun getTrendingMovies(): Flow<MovieList> = mostPopularMovies.map { movieList ->
+        val list = movieList.subList(fromIndex = 0, toIndex = 10).map {
+            Movie.fromMoviesResponseItem(it)
+        }
+        MovieList(value = list)
+    }
+
+    override fun getTop10Movies(): Flow<MovieList> = top250MoviesWithWideThumbnail.map {
+        val list = it.subList(fromIndex = 20, toIndex = 30)
+        MovieList(list)
+    }
+
+    override fun getNowPlayingMovies(): Flow<MovieList> = inTheatersMovies.map { movieList ->
+        val list = movieList
+            .subList(fromIndex = 0, toIndex = 10)
             .map {
-                Movie(
-                    id = it.id,
-                    posterUri = it.image_2_3,
-                    name = it.title,
-                    description = it.fullTitle
-                )
+                Movie.fromMoviesResponseItem(it)
             }
-            .subList(fromIndex = 0, toIndex = 10)
-        emit(MovieList(value = list))
+        MovieList(value = list)
     }
 
-    override fun getMovieCategories() = flowOf(categoryList)
+    override fun getMovieCategories() = categoryList
 
     override suspend fun getMovieCategoryDetails(categoryId: String): MovieCategoryDetails {
-        val category = categoryList.find { it.id == categoryId } ?: categoryList.first()
-        val movieList = top250Movies.shuffled().subList(fromIndex = 0, toIndex = 20)
+        val categoryList = readCategoryList()
+        val movieList = readTop250MovieList()
+
+        val category = categoryList.find { categoryId == it.id } ?: categoryList.first()
+        val list = movieList.shuffled().subList(0, 20).map {
+            Movie.fromMoviesResponseItem(it)
+        }
         return MovieCategoryDetails(
             id = category.id,
             name = category.name,
-            movies = MovieList(movieList)
+            movies = MovieList(list)
         )
     }
 
     override suspend fun getMovieDetails(movieId: String): MovieDetails {
-        val movie = top250Movies.find { it.id == movieId } ?: top250Movies.first()
+        val movieList = readTop250MovieList()
+        val found = movieList.find { it.id == movieId } ?: movieList.first()
+        val movie = Movie.fromMoviesResponseItem(found)
+        val similarMovieList = (1..2).map { Movie.fromMoviesResponseItem(movieList.random()) }
```

Comment:
```suggestion
        val similarMovieList = repeat(2) { Movie.fromMoviesResponseItem(movieList.random()) }
```

Reply:
Thanks for suggestion, and the suggested code sets Unit to similarMovieList if my understanding is correct. 

My original intention is create a list with two random Movie objects. So, I am updating as below:

```
val similarMovieList = movieList.shuffled().subList(0, 2).map {
            Movie.from(it)
        }
```